services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/application/ApplicationServiceITest.kt

```diff
@@ -1423,16 +1431,31 @@ class ApplicationServiceITest : DatabaseTest() {
         }
 
         @Test
-        fun `when application not in Allu should delete application`() {
+        fun `when application not in Allu should delete application and all its attachments`() {
             val hanke = hankeFactory.saveMinimal()
             val application =
                 applicationFactory.saveApplicationEntity(USERNAME, hanke = hanke, alluId = null)
+            // assert that application is saved to database
             assertThat(applicationRepository.findAll()).hasSize(1)
+            val attachment1 = attachmentFactory.saveAttachment(application.id!!)
+            val attachment2 = attachmentFactory.saveAttachment(application.id!!)
+            // assert that attachments are saved to cloud
+            assertTrue(applicationAttachmentContentService.exists(attachment1.blobLocation!!))
+            assertTrue(applicationAttachmentContentService.exists(attachment2.blobLocation!!))
```

Comment:
I think it's sufficient to check that there are 2 attachments. 
```suggestion
            assertThat(fileClient.listBlobs(Container.HAKEMUS_LIITTEET)).hasSize(2)
```

Reply:
Refactored to use fileClient. Removed unnecessary `applicationAttachmentContentService` from this test class.